### PR TITLE
Validate offer affiliate & arbitrator

### PR DESCRIPTION
**Pull request by cuongdo**
This change also makes the makeOffer mutation default to the listing
affiliate and arbitrator when those aren't specified in the offer. A
passed in affiliate or arbitrator may still override the listing
defaults.This allows origin-admin to generate invalid offers (for
testing) but provides a more concise interface for normal offers.

Also adds partial validation for offer commission.

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] Wrap any displayed ETH addresses with [`formattedAddress`](https://github.com/OriginProtocol/origin/blob/master/origin-dapp/src/utils/user.js#L15-L17)
- [ ] Wrap any new text/strings for translation
- [ ] Map any new environment variables with a default value in the Webpack config
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/origin-docs)

### CODE REVIEW

**Comment by cuongdo** [comment]
@nick I've changed validation of the offer arbitrator and made some other fixes. Please take a look.

File: experimental/origin-eventsource/src/index.js [hunk 1]
```diff
@@ -7,12 +7,13 @@ const pick = require('lodash/pick')
...
 
 class OriginEventSource {
-  constructor({ ipfsGateway, marketplaceContract, web3 }) {
+  constructor({ ipfsGateway, marketplaceContract, web3, arbitrator }) {
     this.ipfsGateway = ipfsGateway
     this.contract = marketplaceContract
     this.web3 = web3
     this.offerCache = {}
     this.listingCache = {}
+    this.arbitrator = arbitrator ? arbitrator.toLowerCase() : ZERO_ADDRESS
```
Comment (nick) [inline experimental/origin-eventsource/src/index.js]: Do we still need this?

File: experimental/origin-graphql/src/mutations/marketplace/_validation.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export async function validateNewOffer(listingId, data) {
+  const listing = await contracts.eventSource.getListing(listingId)
+
+  const unitsAvailable = Number(listing.unitsAvailable)
+  const offerQuantity = Number(data.quantity)
+  if (offerQuantity > unitsAvailable) {
+    throw new Error(`Insufficient units available (${unitsAvailable}) for offer (${offerQuantity})`)
+  }
+
+  const depositAvailable = web3.utils.toBN(listing.depositAvailable)
```
Comment (nick) [inline experimental/origin-graphql/src/mutations/marketplace/_validation.js]: Can we refer to contracts.web3 so we don't rely on a web3 global

File: experimental/origin-eventsource/src/index.js (hunk 1 above)
Comment (cuongdo) [reply experimental/origin-eventsource/src/index.js]: removed

File: experimental/origin-graphql/src/mutations/marketplace/_validation.js (hunk 2 above)
Comment (cuongdo) [reply experimental/origin-graphql/src/mutations/marketplace/_validation.js]: I've added a local `web3`

**Pull request merged**